### PR TITLE
Add a `disas` test suite for Wasmtime

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit adds a suite of tests at `tests/disas/*.wat` which is intended to replace the tests in `cranelift/filetests/filetests/wasm`. Tests are configured differently than before using Wasmtime CLI flags rather than a custom TOML-based configuration scheme. Otherwise though the same shape of tests is supported.

This commit migrates a small handful of tests as a showcase and bulk migration is left for a follow-up.

### CODE REVIEW

**Comment by alexcrichton** [comment]
This was inspired by https://github.com/bytecodealliance/wasmtime/pull/8125 and the discussion there

**Review by fitzgen — APPROVED** [review]
Instead of `tests/asm` maybe we should call this `tests/dis` or `tests/disas` since there are CLIF "disassemblies" in there too?

File: tests/asm.rs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+            if let Some(s) = directive.strip_prefix("target: ") {
+                if target.is_some() {
+                    bail!("two targets have been specified");
+                }
+                target = Some(s);
+                continue;
+            }
+
+            bail!("unknown directive: {directive}");
+        }
```
[the remark this answers is not in the document]
Comment (fitzgen) [reply tests/asm.rs:136]: > I was sort of also taking inspiration from compiletest-based tests in rust-lang/rust, but I know the parsing there is sort of gnarly nowadays given all the directives it supports.

Yeah this is the kind of thing I am afraid of

[the remark this answers is not in the document]
Comment (alexcrichton) [reply tests/asm.rs]: Ok, I've switched back to TOML

**Pull request merged**